crates/ethrpc/src/http.rs

```diff
@@ -97,32 +101,103 @@ async fn execute_rpc<T: DeserializeOwned>(
         .send()
         .await
         .map_err(|err: reqwest::Error| {
-            tracing::warn!(name = %inner.name, %id, %err, "failed to send request");
+            if let Some(ref req_id) = request_id {
+                tracing::warn!(
+                    name = %inner.name,
+                    rpc_request_id = %id,
+                    request_id = %req_id,
+                    error = %err,
+                    "failed to send request"
+                );
+            } else {
+                tracing::warn!(
+                    name = %inner.name,
+                    rpc_request_id = %id,
+                    error = %err,
+                    "failed to send request"
+                );
+            }
```

Comment:
No need to these conditions. Logging `None` is totally fine.
```suggestion
            tracing::warn!(
                name = %inner.name,
                rpc_request_id = %id,
                request_id,
                error = %err,
                "failed to send request"
            );
```